zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -355,6 +432,92 @@ impl DiskWriteBatch {
             self.zs_insert(&orchard_tree_cf, height, trees.orchard);
         }
 
+        if let Some(subtree) = trees.sapling_subtree {
+            let (subtree_end, subtree_start) = subtree.bounds();
+
+            let start_height = if subtree_end.index == 0 {
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(Height(1)..)
+                    .next()
+                    .expect("must have sapling notes on disk");
+
+                start_height
+            } else if let Some(start_height) =
+                zebra_db.sapling_subtree_start_height_by_index(subtree_end.index)
+            {
+                start_height
+            } else {
+                let prev_subtree_end_height = zebra_db
+                    .sapling_subtree_by_index(subtree_end.index - 1)
+                    .expect("must have previous subtree")
+                    .end;
+
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(prev_subtree_end_height.next()..)
+                    .next()
+                    .expect("must have previous subtree");
+
+                start_height
```

Comment:
Unfortunately, I can't see how this determines the right start height. Won't `sapling_tree_by_height_range` return the given height?

Reply:
Ah this definitely needs a comment.

So if there isn't an entry at this index with just a block height (the start height, entered when a block completes a subtree and starts a new one), it will look for the height of the first note commitment tree entry after the end height of the previous subtree.

Reply:
Added comments in https://github.com/ZcashFoundation/zebra/pull/7334/commits/aea9320805ed46f7fd58866644b792267592ea7b.